tests/yam/agama/patch_agama.pm

```diff
@@ -0,0 +1,35 @@
+## Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: This module use yupdate patch the Agama on Live Medium
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    assert_screen('agama_product_selection', 120);
+
+    select_serial_terminal;
+
+    my ($repo, $branch) = split /#/, get_var('YUPDATE_GIT');
+    if ($repo && $branch) {
```

Comment:
please leave the control of error to yupdate itself, not need for extra logic, if I would need to read the message "Patch agama SKIP" I wouldn't know what do to...or what does it mean, I would still need to check the patch command (it is repo wrong? it is branch wrong? did I forget to pass this settting?) Besides when doing condition in Perl it is better to not use if/else structure and do something like:
`record_info ("...") unless ($repo && $branch)` which safe spaces, but as  I said, I'd prefer not to have error control, because you have to cover too many cases to help the user, and the patch itself have some defaults values afir.

Reply:
The "yudate patch" command will not give any error or message if you have no any parameter, that's why we need handle this.
localhost:~ # yupdate patch
localhost:~ # echo $?
0

And i think we better not use unless if we need "if else" scenario. See below discussion on unless:
https://stackoverflow.com/questions/3048726/usage-of-if-versus-unless-for-perl-conditionals

Record_info give some confusing, i will update it.

Reply:
The unless discussion goes for each programming languages, it wasn't my point, my point was to give the error/info ealier: `record_info ("...") if !($repo && $branch)` instead of having the structure `if { ... } else { ... }` save lines and characters. It is used a lot in the repo, but it is just code style to look more like Perl and less like C.

We can use yupdate without branch as it default to master afir, so it doesn't make sense to control both parameters mandatory.
A simple solution is just to use `get_required_var`, due to we are going to patch always the livecd, test will be in a different repo than the agama project, where the 3 test for CI are packaged in the livecd. In other words, it will not be skip by design.